### PR TITLE
docs(update-guide): Guide to update Tasklist/Operate versions

### LINKED ISSUE #592: Document update procedure for Operate
We need to raise the important requirements to update operate version.
Right now, we have customers, for example, jumping from version `1.0` to `1.3`, which shouldn't be a problem overall, except they need to import all the data from version `1.0` (wait for operate importer to finish) and then be able to update to `1.3`

This should be clearly specified in both On-Prem and CC.

Also, Operate importer currently support only importing data from 2 minor versions, meaning the current one and the one before.
For example, cluster is in version `1.4`, it can import data from `1.3` AND `1.4`

### PR DESCRIPTION
**Pull request by santanaluiz**
Guide to update Tasklist/Operate versions
Both on-prem and cloud

Closes #592

### CODE REVIEW

**Comment by akeller** [comment]
@santanaluiz is this part of release 1.4 or should we merge immediately after this is reviewed?

**Comment by santanaluiz** [comment]
@akeller we can merge already. That's valid for current versions

File: docs/self-managed/operate-deployment/updating-operate.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Let's assume the following scenario:
+A server running Operate version `1.0` wants to update to version `1.3`.
+
+We recommend to follow the steps below:
+
+### Progressively update
+
+1. Update Operate and Zeebe to version `1.1`
+2. Let it run for some hours and verify if everything works as expected
```
Comment (christinaausley) [inline docs/self-managed/operate-deployment/updating-operate.md:22]: What does "it" stand for here?

File: docs/self-managed/operate-deployment/updating-operate.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+3. Repeat steps `1` and `2` for version `1.2`
+4. Then update both to version `1.3` safely
```
Comment (christinaausley) [inline docs/self-managed/operate-deployment/updating-operate.md:24]: Update both what here? Operate and Zeebe, or the two versions? How do you do this safely?

File: docs/self-managed/operate-deployment/updating-operate.md (hunk 1 above)
Comment (christinaausley) [inline docs/self-managed/operate-deployment/updating-operate.md:22]: Approximately how many hours? Two, 24?

File: versioned_docs/version-1.1/components/operate/deployment/updating-operate.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
```
[the remark this answers is not in the document]
Comment (santanaluiz) [reply versioned_docs/version-1.1/components/operate/deployment/updating-operate.md:1]: added to 1.1 by mistake... moving it

File: docs/self-managed/operate-deployment/updating-operate.md (hunk 1 above)
Comment (santanaluiz) [reply docs/self-managed/operate-deployment/updating-operate.md]: Operate, changing docs. Please check again

File: docs/self-managed/operate-deployment/updating-operate.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Let's assume the following scenario:
+A server running Operate version `1.0` wants to update to version `1.3`.
+
+We recommend to follow the steps below:
+
+### Progressively update
+
+1. Update Operate and Zeebe to version `1.1`
+2. Let it run for some hours and verify if everything works as expected
+3. Repeat steps `1` and `2` for version `1.2`
```
[the remark this answers is not in the document]
Comment (santanaluiz) [reply docs/self-managed/operate-deployment/updating-operate.md]: Correct, but I rephrased

File: versioned_docs/version-1.1/components/operate/deployment/updating-operate.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (santanaluiz) [reply versioned_docs/version-1.1/components/operate/deployment/updating-operate.md:1]: I moved to user-guide, but TBH doesn't seem like the right place. I wanted to have this documentation for both cloud and self managed.

Any suggestions here?

[the remark this answers is not in the document]
Comment (christinaausley) [reply versioned_docs/version-1.1/components/operate/deployment/updating-operate.md:1]: If these instructions are specific only to Self-Managed, this is where the docs should go.

If these instructions are applicable to both Cloud and Self-Managed, these instructions should be placed within Cloud, likely within the userguides as you have them, and can be linked to within Self-Managed. Does this make sense? We'd prefer to not have duplicate documentation.

**Comment by akeller** [comment]
Is this good to merge?

**Comment by santanaluiz** [comment]
@christinaausley can you double check?

@akeller waiting for review

**Comment by christinaausley** [comment]
@santanaluiz @akeller Just following up here on where the changes are meant to be made -- it looks like we have these changes in the Cloud user guide, Self-Managed, and versioned docs.

Should these changes be made solely in the user guide and referenced to/linked to within Self-Managed? If so, we need to remove the adjustments in Self-Managed and the versioned docs and keep these changes within the user guide.

**Comment by akeller** [comment]
> @santanaluiz @akeller Just following up here on where the changes are meant to be made -- it looks like we have these changes in the Cloud user guide, Self-Managed, and versioned docs.
> 
> Should these changes be made solely in the user guide and referenced to/linked to within Self-Managed? If so, we need to remove the adjustments in Self-Managed and the versioned docs and keep these changes within the user guide.

I'll add this to my list to carefully review tomorrow, not sure I can give you a quick answer with just a skim.

**Comment by santanaluiz** [comment]
@akeller @christinaausley  From my side, I don't have a strong opinion about making in just one point and referencing.

It would probably make life easier in case of updating it later, so it would be in a single point

**Comment by christinaausley** [comment]
@santanaluiz Excellent! Could you please revert the versioned docs to their original form and remove these adjustments within Self-Managed so we are solely adding the two userguide docs? I'm providing a link to these docs in Self-Managed in my latest commit.

Once this is complete, we should be good to merge.

**Comment by akeller** [comment]
Hi folks - we need to get this PR merged this week. If no action is taken by tomorrow end of my day, I will make changes at my discretion and merge to `master` on Friday. 

FYI @christinaausley @santanaluiz

**Comment by akeller** [comment]
Following up - given Luiz is out on vacation, I'll take care of the adjustments myself and merge on Friday.

**Pull request merged**